hax-lib-macros/src/lib.rs

```diff
@@ -690,3 +690,141 @@ macro_rules! make_quoting_proc_macro {
 make_quoting_proc_macro!(fstar(fstar_expr, fstar_before, fstar_after, fstar_replace, hax_backend_fstar)
                          coq(coq_expr, coq_before, coq_after, coq_replace, hax_backend_coq)
                          proverif(proverif_expr, proverif_before, proverif_after, proverif_replace, hax_backend_proverif));
+
+/// Turns a newtype `struct Foo(SomeType);` into a refinement
+/// type. The struct should have exacty one unnamed private field.
+///
+/// This macro takes one argument: a boolean predicate on that refines
+/// values of type `SomeType`.
+///
+/// For example, the following type defines bounded `u64` integers.
+///
+/// ```
+/// #[hax_lib::refinement_type(|x| x >= MIN && x <= MAX)]
+/// pub struct BoundedU64<const MIN: u64, const MAX: u64>(u64);
```

Comment:
Why is `MIN` and `MAX` needed here when it's only needed for the refinement?

Reply:
It's an example only, it will be made into an actual lib in #642